src/plugins/intel_gpu/src/plugin/ops/gather.cpp

```diff
@@ -103,8 +103,9 @@ void CreateGatherOpBase(ProgramBuilder& p, const std::shared_ptr<T>& op, const i
     auto indices_constant = ov::as_type_ptr<ov::op::v0::Constant>(indices_node);
     bool is_indices_constant = (indices_constant) ? true : false;
 
-    if (is_static && axis == 0 && input_rank > 1 && indices.get_partial_shape().rank().get_length() == 0 &&
-        std::equal(input_shape.begin()+1, input_shape.end(), out_shape.begin()+1) && is_indices_constant) {
+    if (is_static && axis == 0 && is_indices_constant && input_rank > 1 &&
+        indices.get_partial_shape().rank().get_length() == 0 && 
+        std::equal(input_shape.begin()+1, input_shape.end(), out_shape.begin()+1, out_shape.end())) {
```

Comment:
* could you add a test case for this?
* could you add description following the code review template?